app/services/ipam/base.rb

```diff
@@ -1,23 +1,40 @@
 module IPAM
   class Base
+    extend ActiveModel::Naming
+    extend ActiveModel::Translation
+
     delegate :logger, :to => :Rails
     attr_accessor :mac, :subnet, :excluded_ips
+    attr_reader :errors
 
     def initialize(opts = {})
       @subnet = opts[:subnet]
       @excluded_ips = opts.fetch(:excluded_ips, [])
       @mac = opts.fetch(:mac, nil)
+      @errors = ActiveModel::Errors.new(self)
 
       normalize_mac!
     end
 
+    def suggest_new?
+      true
+    end
+
+    def serializable_hash(options = nil)
+      {'suggest_new' => suggest_new?}
+    end
+
     private
 
     def normalize_mac!
       return unless self.mac.present?
       self.mac = Net::Validations.normalize_mac(self.mac)
     rescue Net::Validations::Error
-      raise Foreman::Exception.new(N_("'%s' is not a valid MAC address.") % self.mac)
+      errors.add(:mac, _('is not a valid MAC address'))
+    end
+
+    def read_attribute_for_validation(attr)
```

Comment:
I had to look up what this does, mind to add a comment explaining why this is ncessary?